cdap-distributions/src/emr/cdap-conf.json

```diff
@@ -2,37 +2,36 @@
   "cdap": {
     "cdap_site": {
       "dashboard.bind.port": "11011",
-      "dataset.executor.container.memory.mb": "1536",
       "explore.enabled": "{{EXPLORE_ENABLED}}",
-      "explore.executor.container.memory.mb": "2304",
       "http.client.read.timeout.ms": "120000",
       "kafka.log.dir": "/mnt/cdap/kafka-logs",
-      "log.saver.run.memory.megs": "1536",
-      "master.service.memory.mb": "1536",
       "master.startup.checks.enabled": "false",
-      "metrics.memory.mb": "1536",
-      "metrics.processor.memory.mb": "1536",
+      "enable.unrecoverable.reset": "true",
       "router.bind.port": "11015",
-      "stream.container.memory.mb": "1536",
-      "twill.java.reserved.memory.mb": "350",
-      "router.server.address": "{{ROUTER_IP_ADDRESS}}",
-      "zookeeper.quorum": "{{ZK_QUORUM}}"
+      "router.server.address": "{{ROUTER_IP_ADDRESS}}"
     },
     "cdap_env": {
       "explore_enabled": "true",
       "kerberos_enabled": "false",
       "hadoop_conf_dir": "/etc/hadoop/conf",
       "hadoop_home_warn_suppress": "$(hadoop dfsadmin -safemode wait >/dev/null 2>&1; sleep 1; [[ ${UID} -eq 0 ]] && su - hdfs -c 'for d in /cdap /user/cdap; do hdfs dfs -mkdir -p ${d} ; hdfs dfs -chown cdap ${d}; done' >/dev/null 2>&1; echo true)",
-      "hive_home": "/usr/lib/hive",
+      "hbase_home": "$(until $(which hbase &>/dev/null); do sleep 5; done; echo /usr/lib/hbase)",
+      "hive_home": "$(until $(which hive &>/dev/null); do sleep 5; done; echo /usr/lib/hive)",
       "hive_conf_dir": "/etc/hive/conf",
-      "hive_classpath": "{{HIVE_CLASSPATH}}",
-      "hive_exec_engine": "{{HIVE_EXEC_ENGINE}}",
+      "hive_classpath": "${HIVE_CLASSPATH:-/etc/hive/conf:$(ls -1 /usr/lib/hive/lib/*.jar | tr '\\n' ':')$(if [[ -r /etc/hive/conf/hive-env.sh ]]; then source /etc/hive/conf/hive-env.sh ; fi ; if [[ -d ${HIVE_AUX_JARS_PATH} ]]; then ls -1 ${HIVE_AUX_JARS_PATH}/*.jar | tr '\\n' ':'; else echo ${HIVE_AUX_JARS_PATH} | tr ',' ':'; fi)$(ls -1 /usr/lib/hive/auxlib/*.jar 2>/dev/null | tr '\\n' ':')}",
```

Comment:
should this have the ``until`` loop also?  or are we depending on ``hive_home`` being processed first?  (i'm not sure if order is guaranteed...)